WordPress/gutenberg
# Try using variable for root padding value

## What?


* Adds a setting to `theme.json` so themes can opt in to the new behaviour of root padding
* Adds a default non-zero value for root padding
* Outputs that value as a custom property on the root element
* Allows full-width blocks to compensate for that value with negative margins

## Why?


Partly fixes #35607 - only addresses the root block part of the equation. #39871 deals with full width in nested blocks.
Fixes #35884 for themes that opt into this change. 

Themes shouldn't have to add their own rules to make full width alignments work properly.

## How?


First, we're setting a default padding value in theme.json. It can be overridden by any user-set padding in the global styles interface.

Then, on the server side, we set top and bottom padding to the root container, but left and right padding to the direct children of root. This is so that those children still stretch full width as expected, but _their_ children get the effects of the padding.

On the site editor side, the direct children of root container get the left/right padding applied, so they still appear as full-width. (If we wanted to only apply full-width to some of the root children, maybe we could add padding to the root and enable the align control on them so they could be set manually? And add negative margins to `alignfull` blocks)

On the post editor side, the root container also gets its left/right padding, and any `alignfull` children are fixed with negative margins.


## Testing Instructions





0. Add `"useRootVariables": true` to the `settings` of your theme's `theme.json` to enable these changes.
1. Add a few layout blocks (e.g. Group) to the post and the site editor
2. Make some of those blocks full width
3. Check that there's a default padding value in both editors and front end, and full width blocks behave as expected
4. Change root padding value in the site editor by going to Global Styles > Layout > Padding
5. Check that new padding value works in both editors and front end.
6. Check that removing `"useRootVariables": true` from `theme.json` reverts to the previous behaviour.

## Screenshots or screencast 
![ezgif com-gif-maker (5)](https://user-images.githubusercontent.com/8096000/160996605-fcc4d868-1c06-434d-85dd-a94f4b29bb1f.gif)
